### PR TITLE
[Exp PyROOT] Raise detailed error when Python version is different fr…

### PR DESCRIPTION
**Pull request by maxgalli**
…om the one used to build libcppyy

Suggested by Chris Burr

### CODE REVIEW

File: bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py [hunk 1]
```diff
@@ -55,7 +55,15 @@
 major, minor = sys.version_info[0:2]
 py_version_str = '{}_{}'.format(major, minor)
 libcppyy_mod_name = 'libcppyy' + py_version_str
-importlib.import_module(libcppyy_mod_name)
+try:
+    importlib.import_module(libcppyy_mod_name)
+except ModuleNotFoundError as err:
```
Comment (etejedor) [inline bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py:60]: Please check if this is `ImportError` in Python2, and if there is a way to do this in a unified manner for both Python versions.

File: bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py [hunk 2]
```diff
@@ -55,7 +55,15 @@
...
+    if err.name.startswith('libcppyy'):
+        raise ModuleNotFoundError(
+                'Failed to import {}. Probably ROOT has not been built for Python {}.{}'.format(
```
Comment (etejedor) [inline bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py:63]: Instead I would say: `Please check that ROOT has been built for Python...`

File: bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py [hunk 3]
```diff
@@ -55,7 +55,15 @@
 major, minor = sys.version_info[0:2]
 py_version_str = '{}_{}'.format(major, minor)
 libcppyy_mod_name = 'libcppyy' + py_version_str
-importlib.import_module(libcppyy_mod_name)
+try:
+    importlib.import_module(libcppyy_mod_name)
+except ModuleNotFoundError as err:
+    if err.name.startswith('libcppyy'):
```
Comment (etejedor) [inline bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py:61]: I would not add this extra check here, if the import fails we error out and we provide a suggestion to the user about what can be going on.

File: bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py (hunk 1 above)
Comment (maxgalli) [reply bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py:60]: yes, it's ```ImportError``` in Python2
```
Python 2.7.17 (default, Apr 15 2020, 17:20:14) 
[GCC 7.5.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> 
>>> 
>>> import ROOT
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/massimiliano/root-devel/tries/lib/ROOT/__init__.py", line 28, in <module>
    import cppyy
  File "/home/massimiliano/root-devel/tries/lib/cppyy/__init__.py", line 58, in <module>
    importlib.import_module(libcppyy_mod_name)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named libcppyy2_7
```

changing

File: bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py [hunk 4]
```diff
@@ -55,7 +55,20 @@
...
+else:
+    CustomImportError = ImportError
+
+try:
+    importlib.import_module(libcppyy_mod_name)
+except CustomImportError as err:
+    if not hasattr(err, 'name'):
+        err.name = err.args[0].split()[-1]
+    raise CustomImportError(
+            'Failed to import {}. Please check that ROOT has been built fot Python {}.{}'.format(
```
Comment (maxgalli) [inline bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py:70]: typo

File: bindings/pyroot_experimental/cppyy/patches/multi_python_cppyy.patch [hunk 5]
```diff
@@ -5,20 +5,20 @@ index 2f39146505..3653596167 100644
 @@ -756,12 +756,6 @@ static PyMethodDef gCPyCppyyMethods[] = {
      {nullptr, nullptr, 0, nullptr}
  };
- 
+
```
Comment (maxgalli) [inline bindings/pyroot_experimental/cppyy/patches/multi_python_cppyy.patch:8]: update patch

**Pull request merged**